*** Cmd:    /bin/sh check-sources.sh
*** Date:   Fri  5 May 21:10:02 BST 2017
*** Source: http://github.com/tomstdenis/libtomcrypt/tree/develop
            http://www.libtom.net/LibTomCrypt

*** Libtomcrypt repo: heads/develop 1.17-376-g4981e2a

*** diff fortuna.c:
--- ../libtomcrypt/src/prngs/fortuna.c	2017-05-05 21:08:45.565297598 +0100
+++ ./fortunad/ltc_fortuna.c	2017-04-30 14:59:17.966913318 +0100
@@ -320,6 +320,7 @@
 {
    int         x, err;
    hash_state *md;
+   hash_state smd;                           /* patched */
 
    LTC_ARGCHK(out    != NULL);
    LTC_ARGCHK(outlen != NULL);
@@ -334,7 +335,8 @@
       return CRYPT_BUFFER_OVERFLOW;
    }
 
-   md = XMALLOC(sizeof(hash_state));
+   /* md = XMALLOC(sizeof(hash_state)); */   /* patched */
+   md = &smd ;                               /* patched */
    if (md == NULL) {
       LTC_MUTEX_UNLOCK(&prng->fortuna.prng_lock);
       return CRYPT_MEM;
@@ -367,10 +369,10 @@
    err = CRYPT_OK;
 
 LBL_ERR:
-#ifdef LTC_CLEAN_STACK
+/* #ifdef LTC_CLEAN_STACK */                 /* patched */
    zeromem(md, sizeof(*md));
-#endif
-   XFREE(md);
+/* #endif */                                 /* patched */
+/* XFREE(md); */                             /* patched */
    LTC_MUTEX_UNLOCK(&prng->fortuna.prng_lock);
    return err;
 }

*** diff tomcrypt_prng.h:
--- ../libtomcrypt/src/headers/tomcrypt_prng.h	2017-05-05 21:08:45.541298342 +0100
+++ ./headers/tomcrypt_prng.h	2017-04-30 14:59:17.862916513 +0100
@@ -189,7 +189,7 @@
  */
 unsigned long rng_get_bytes(unsigned char *out,
                             unsigned long outlen,
-                            void (*callback)(void));
+                            void (*callback)(void*), void *desc);
 
 int rng_make_prng(int bits, int wprng, prng_state *prng, void (*callback)(void));
 

*** diff tomcrypt_custom.h:
--- ../libtomcrypt/src/headers/tomcrypt_custom.h	2017-05-05 21:08:45.541298342 +0100
+++ ./headers/tomcrypt_custom.h	2017-04-30 14:59:17.862916513 +0100
@@ -1,6 +1,8 @@
 #ifndef TOMCRYPT_CUSTOM_H_
 #define TOMCRYPT_CUSTOM_H_
 
+#include "tomcrypt_nim.h"
+
 /* macros for various libc functions you can change for embedded targets */
 #ifndef XMALLOC
    #ifdef malloc

*** End
